### PR TITLE
feat: floating widget &  variable resolve

### PR DESCRIPTION
**Pull request by ritvikmahajan17**
# Description

1. Floating scoped variables widget in base deployment template and env override

Screenshot 2023-10-13 at 12 40 28 PM

2. Functionality to resolve scoped variables to corresponding values in diff view mode in  a) Deployment History  b) Build & deploy 

Before:
Screenshot 2023-10-13 at 12 42 50 PM

After:
Screenshot 2023-10-13 at 12 42 53 PM

Fixes #4095

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx [hunk 1]
```diff
@@ -60,6 +64,13 @@ export default function DeploymentHistoryDiffView({
         )
     }
 
+    const handleShowVariablesClick = () => {
+        console.log('handleShowVariablesClick')
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx:68]: please remove console.log

Comment (ritvikmahajan17) [reply src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx]: resolved

File: src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx [hunk 2]
```diff
@@ -137,8 +160,18 @@ export default function DeploymentHistoryDiffView({
                         className="code-editor-header-value pl-16 pr-16 pt-12 pb-12 fs-13 fw-6 cn-9 bcn-0 dc__top-radius-4 dc__border-bottom"
                         data-testid="configuration-link-comparison-body-heading"
                     >
-                        {baseTemplateConfiguration?.codeEditorValue?.['displayName']}
+                        <span>{baseTemplateConfiguration?.codeEditorValue?.['displayName']}</span>
+                        {isVariablesAvailable ? (
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx:164]: we can reWrite this as `{isVariablesAvailable && (`

File: src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx [hunk 3]
```diff
@@ -337,6 +363,12 @@ export default function TriggerViewConfigDiff({
         )
     }
 
+    const handleShowVariablesClick = () => {
+        // console.log('handleShowVariablesClick')
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx:367]: Please remove this commented code

File: src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx [hunk 4]
```diff
@@ -400,8 +432,17 @@ export default function TriggerViewConfigDiff({
                 <div className="p-16 dc__overflow-scroll">
                     {renderConfigValuesDiff()}
                     <div className="en-2 bw-1 br-4">
-                        <div className="code-editor-header-value flex left pt-8 pb-8 pl-16 pr-16 fs-13 fw-6 lh-20 cn-9 bcn-0 dc__top-radius-4 dc__border-bottom">
-                            {editorValues.displayName}
+                        <div className="code-editor-header-value left pt-8 pb-8 pl-16 pr-16 fs-13 fw-6 lh-20 cn-9 bcn-0 dc__top-radius-4 dc__border-bottom">
+                            <span>{editorValues.displayName}</span>
+                            {isVariablesAvailable ? (
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx:437]: Please rewrite this as `{isVariablesAvailable && (`

Comment (ritvikmahajan17) [reply src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx]: resolved

File: src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx (hunk 3 above)
Comment (ritvikmahajan17) [reply src/components/app/details/triggerView/triggerViewConfigDiff/TriggerViewConfigDiff.tsx]: resolved

File: src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx (hunk 2 above)
Comment (ritvikmahajan17) [reply src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDiffView.tsx]: resolved

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**